compositor_render/src/state/render_loop.rs

```diff
@@ -91,31 +91,52 @@ pub(super) fn read_outputs(
                     })
                 }
             },
-            None => {
-                let (y, u, v) = RGBColor::BLACK.to_yuv();
-                ctx.wgpu_ctx.utils.fill_r8_with_value(
-                    ctx.wgpu_ctx,
-                    output.output_texture.yuv_textures().plane(0),
-                    y,
-                );
-                ctx.wgpu_ctx.utils.fill_r8_with_value(
-                    ctx.wgpu_ctx,
-                    output.output_texture.yuv_textures().plane(1),
-                    u,
-                );
-                ctx.wgpu_ctx.utils.fill_r8_with_value(
-                    ctx.wgpu_ctx,
-                    output.output_texture.yuv_textures().plane(2),
-                    v,
-                );
+            None => match output.output_format {
+                OutputFrameFormat::PlanarYuv420Bytes => {
+                    let (y, u, v) = RGBColor::BLACK.to_yuv();
+                    ctx.wgpu_ctx.utils.fill_r8_with_value(
+                        ctx.wgpu_ctx,
+                        output.output_texture.yuv_textures().plane(0),
+                        y,
+                    );
+                    ctx.wgpu_ctx.utils.fill_r8_with_value(
+                        ctx.wgpu_ctx,
+                        output.output_texture.yuv_textures().plane(1),
+                        u,
+                    );
+                    ctx.wgpu_ctx.utils.fill_r8_with_value(
+                        ctx.wgpu_ctx,
+                        output.output_texture.yuv_textures().plane(2),
+                        v,
+                    );
 
-                let pending_download = output.output_texture.start_download(ctx.wgpu_ctx);
-                partial_textures.push(PartialOutputFrame::PendingYuvDownload {
-                    output_id: output_id.clone(),
-                    pending_download,
-                    resolution: output.output_texture.resolution().to_owned(),
-                });
-            }
+                    let pending_download = output.output_texture.start_download(ctx.wgpu_ctx);
+                    partial_textures.push(PartialOutputFrame::PendingYuvDownload {
+                        output_id: output_id.clone(),
+                        pending_download,
+                        resolution: output.output_texture.resolution().to_owned(),
+                    });
+                }
+                OutputFrameFormat::RgbaWgpuTexture => {
+                    let resolution = output.output_texture.resolution();
+                    let rgba_texture = RGBATexture::new(ctx.wgpu_ctx, resolution);
+                    ctx.wgpu_ctx.utils.fill_r8_with_value(
+                        ctx.wgpu_ctx,
+                        rgba_texture.texture(),
+                        0.0,
+                    );
+                    let wgpu_texture = rgba_texture.texture().copy_wgpu_texture(ctx.wgpu_ctx);
```

Comment:
just extract the texture from from rgba_texture you don't need to copy it

Reply:
I would need to either:
1. Make it pub
```rust
pub struct RGBATexture(pub Texture);
```
2. Add method returning texture
```rust
    pub fn texture(&self) -> &Texture {
        &self.0
    }

    pub fn texture_owned(self) -> Texture {
        self.0
    }
```

I can't do
```rust
let wgpu_texture = rgba_texutre.texture().texture
```
since `texture()` returns `&compositor_render::wgpu::texture::base::Texture` and I can't move out `wgpu::Texture` out of ref.

Reply:
yes In know. I would go with option 2. Not sure about the name, but don't have better idea